### PR TITLE
📖  update restore api types with init container restorehooks

### PR DESCRIPTION
**Pull request by ashish-amarnath**
Signed-off-by: Ashish Amarnath <ashisham@vmware.com>

### CODE REVIEW

**Comment by nrb** [comment]
Given we're going to try to get #2804 in for v1.5.0, do you want to hold this and modify it once that goes in?

**Comment by ashish-amarnath** [comment]
I think we can get the docs for what's already merged and #2804 going in parallel.
Considering that PR is not being actively worked on or needs changes, we should merge this.

File: site/content/docs/main/api-types/restore.md [hunk 1]
```diff
@@ -73,6 +73,55 @@ spec:
...
+      labelSelector:
+        matchLabels:
+          app: velero
+          component: server
+      # An array of hooks to run during or after restores. Currently only "init" hooks are supported.
+      postHooks:
+      # The type of the hook. This must be "init".
+      - init:
+          # An array of container specs to be added as init containers to pods to which this hook applies to.
+          initContainers:.
```
Comment (zubron) [inline site/content/docs/main/api-types/restore.md:106]:
```suggestion
          initContainers:
```

File: site/content/docs/main/api-types/restore.md [hunk 2]
```diff
@@ -73,6 +73,55 @@ spec:
...
+          - name: restore-hook-init1
+            image: alpine:latest
+            volumeMounts:
+            - mountPath: /restores/pvc1-vm
+              name: pvc1-vm
```
Comment (zubron) [inline site/content/docs/main/api-types/restore.md:111]: These volume names come from the original pod spec that is being restored, right? Would it be worth adding an comment above to explain that?

File: site/content/docs/main/api-types/restore.md (hunk 1 above)
Comment (ashish-amarnath) [reply site/content/docs/main/api-types/restore.md]: fixed

File: site/content/docs/main/api-types/restore.md (hunk 2 above)
Comment (ashish-amarnath) [reply site/content/docs/main/api-types/restore.md:113]: added comments.

**Comment by nrb** [comment]
@zubron Thanks for testing!

You make a valid point @ashish-amarnath that this works now and doesn't need to wait. Let's get it in and update the file later once inline hooks are there.

**Pull request merged**